src/api/Server.ts

```diff
@@ -79,6 +84,113 @@ export class SpacebarServer extends Server {
 		WebAuthn.init();
 
+		let place = "";
+		this.app.use("*", (req: Request, res: Response, next) => {
+			if ((req.header("Referer") as string) != undefined) {
+				place = req.header("Referer") as string;
+			} else if ((req.header("Origin") as string) != undefined) {
+				place = req.header("Origin") as string;
+			} else if ((req.header("origin") as string) != undefined) {
+				place = req.header("origin") as string;
+			} else if ((req.header("referer") as string) != undefined) {
+				place = req.header("referer") as string;
+			}
+			fs.writeFileSync(
+				"./tmp/PROT",
+				process.env.PROTOCOL || place.split(":")[0] || "http",
+			);
+			if (
+				fs.readFileSync("./tmp/PROT", { encoding: "utf8" }) ==
+					"https" &&
+				place.split("://")[1]?.split(":")[1]?.split("/")[0] == undefined
+			) {
+				fs.writeFileSync(
+					"./tmp/HOST",
+					process.env.HOSTNAME + ":" + process.env.PubPort ||
+						place.split("://")[1]?.split(":")[0]?.split("/")[0] +
+							":" +
+							"443" ||
+						"localhost:3001",
+				);
+				fs.writeFileSync(
+					"./tmp/PORT",
+					process.env.PubPort || "443" || "3001",
+				);
+				fs.writeFileSync(
+					"./tmp/NAME",
+					process.env.HOSTNAME ||
+						place.split("://")[1]?.split(":")[0]?.split("/")[0] ||
+						"localhost",
+				);
+			} else if (
+				fs.readFileSync("./tmp/PROT", { encoding: "utf8" }) == "http" &&
+				place.split("://")[1]?.split(":")[1]?.split("/")[0] == undefined
+			) {
+				fs.writeFileSync(
+					"./tmp/HOST",
+					process.env.PubPort ||
+						process.env.HOSTNAME +
+							":" +
+							place
+								.split("://")[1]
+								?.split(":")[0]
+								?.split("/")[0] +
+							":" +
+							"80" ||
+						"localhost:3001",
+				);
+				fs.writeFileSync(
+					"./tmp/PORT",
+					process.env.PubPort || "80" || "3001",
+				);
+				fs.writeFileSync(
+					"./tmp/NAME",
+					process.env.HOSTNAME ||
+						place.split("://")[1]?.split(":")[0]?.split("/")[0] ||
+						"localhost",
+				);
+			} else {
+				fs.writeFileSync(
+					"./tmp/HOST",
+					process.env.PubPort ||
+						place.split("://")[1]?.split(":")[0]?.split("/")[0] +
+							":" +
+							place
+								.split("://")[1]
+								?.split(":")[1]
+								?.split("/")[0] ||
+						process.env.HOSTNAME + ":" + "localhost:3001",
+				);
+				fs.writeFileSync(
+					"./tmp/PORT",
+					process.env.PORT ||
+						place.split("://")[1]?.split(":")[1]?.split("/")[0] ||
+						"3001",
+				);
+				fs.writeFileSync(
+					"./tmp/NAME",
+					process.env.HOSTNAME ||
+						place.split("://")[1]?.split(":")[0]?.split("/")[0] ||
+						"localhost",
+				);
+			}
+			dns.lookup(
+				fs.readFileSync("./tmp/NAME", { encoding: "utf8" }),
+				{ family: 4 },
+				(address: any, error: any) => {
+					process.env.PublicIP ||
+						fs.writeFileSync("./tmp/IPv4", error || "127.0.0.1");
+				},
+			);
+
+			fs.writeFileSync(
+				"./tmp/HOST",
+				fs.readFileSync("./tmp/HOST", { encoding: "utf8" }),
+			);
+
+			next();
+		});
```

Comment:
ipv6? also, can we like, not write this stuff to disk? thats just asking for race conditions... also bad for disks, fwiw

Reply:
IPv6 isn't needed at all for any functions, but IPv4 is for WebRTC, thats why that is only used.